#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()
gen.add("max_size", int_t, 0, "the max number of the points of each cluster", 25000, 0, 50000)
gen.add("min_size", int_t, 0, "the minimum number of the points of each cluster", 20, 0, 10000)
gen.add("number_of_neighbors", int_t, 0, "the number of the neighbors used in kdtree search", 10, 0, 1000)
gen.add("smoothness_threshold", double_t, 0, "smoothness threshold of region growing segmentation", 90 / 180.0 * pi, 0, 2 * pi)
gen.add("curvature_threshold", double_t, 0, "curvature threshold of region growing segmentation", 0.1, 0, 1.0)

exit (gen.generate (PACKAGE, "jsk_pcl_ros", "RegionGrowingSegmentation"))
